pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/ResultHolderFactory.java

```diff
@@ -30,7 +30,20 @@
   private ResultHolderFactory() {
   }
 
-  public static final int MAX_INITIAL_RESULT_HOLDER_CAPACITY = 10000;
+  public static final int MAX_INITIAL_RESULT_HOLDER_CAPACITY = 10_000;
+  public static final int MAX_NUM_GROUPS_ALLOWED;
+
+  // Read the max number of groups allowed from unix env variable.
+  static {
+    int numGroups;
+    try {
+      numGroups =
+          Integer.parseInt(System.getProperty("pinot.max.aggr.groups"));
```

Comment:
Yes, agreed. But we currently do not have an elegant solution for passing the server config from server starter all they way down to lowest level apis of query execution. So far the approach has been to modify/corrupt the interfaces to plumb through the settings.

Solving this is outside the scope of this PR. Once we have a solution for that, I'll move this into server config.